### PR TITLE
Task 4

### PR DESCRIPTION
**Pull request by kmaric69**
I'm submitting the fourth assignment.

### CODE REVIEW

File: 04.DOM_manipulation/src/domManipulation.js [hunk 1]
```diff
@@ -1,3 +1,26 @@
...
+    selectedElement.prepend(newElement);
+  },
+  after: function(selectedElement, newElement) {
+    selectedElement.insertAdjacentElement('afterend', newElement);
+  },
+  before: function(selectedElement, newElement) {
+    selectedElement.insertAdjacentElement('beforebegin', newElement);
+  },
+  val: function(selectedElement) {
+    if(selectedElement.nodeName === 'SELECT'){
```
Comment (badurinantun) [inline 04.DOM_manipulation/src/domManipulation.js]: Could you rather check if element is an instace of `HTMLSelectElement`? I think `nodeName` is something XML specific and could vary on different browsers (possible, not probable)

Also add space after the `if` keyword

File: 04.DOM_manipulation/src/domManipulation.js [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+      return selectedElement.options[selectedElement.selectedIndex].value;
+    } else if(selectedElement.nodeName === null) {
```
[the remark this answers is not in the document]
Comment (kmaric69) [reply 04.DOM_manipulation/src/domManipulation.js]: it would be `undefined`

File: 04.DOM_manipulation/src/domManipulation.js (hunk 1 above)
Comment (kmaric69) [reply 04.DOM_manipulation/src/domManipulation.js]: `nodeName` is fully supported in all major browsers except for Internet Explorer (Compatibility unknown) but it's true that it varies from browser to browser how they load XML and handle white spaces and new lines. It's better to use `HTMLSelectElement` in this case.

I will fix it and commit the changes.

File: 04.DOM_manipulation/src/domManipulation.js [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+    selectedElement.prepend(newElement);
+  },
+  after: function(selectedElement, newElement) {
+    selectedElement.insertAdjacentElement('afterend', newElement);
+  },
+  before: function(selectedElement, newElement) {
+    selectedElement.insertAdjacentElement('beforebegin', newElement);
+  },
+  val: function(selectedElement) {
+    if(selectedElement.tagName === 'SELECT') {
```
Comment (badurinantun) [inline 04.DOM_manipulation/src/domManipulation.js]:
```suggestion
    if (selectedElement.tagName === 'SELECT') {
```
Add spaces after if statements

File: 04.DOM_manipulation/src/domManipulation.js [hunk 4]
```diff
@@ -1,3 +1,26 @@
...
+      return selectedElement.options[selectedElement.selectedIndex].value;
+    } else if(selectedElement.tagName === undefined) {
```
Comment (badurinantun) [inline 04.DOM_manipulation/src/domManipulation.js]:
```suggestion
    } else if (selectedElement.tagName === undefined) {
```

File: 04.DOM_manipulation/src/domManipulation.js [hunk 5]
```diff
@@ -1,3 +1,26 @@
-var dom = function(){
-  // code goes here
+var dom = {
```
Comment (badurinantun) [inline 04.DOM_manipulation/src/domManipulation.js]:
```suggestion
const dom = {
```

**Pull request merged**